org_fedora_oscap/rule_handling.py

```diff
@@ -752,3 +782,258 @@ def revert_changes(self, ksdata, storage):
 
         self._kdump_enabled = None
         self._kdump_default_enabled = None
+
+
+class FirewallRules(RuleHandler):
+    """Simple class holding data from the rules affecting firewall configurations."""
+
+    def __init__(self):
+        """Constructor setting the initial value of attributes."""
+
+        self._add_svcs = set()
+        self._remove_svcs = set()
+        self._add_trusts = set()
+        self._add_ports = set()
+
+        self._added_svcs = set()
+        self._added_ports = set()
+        self._added_trusts = set()
+        self._removed_svcs = set()
+
+        self._firewall_enabled = None
+        self._firewall_default_enabled = None
+
+    def add_services(self, services):
+        """
+        Services that should be allowed through firewall.
+
+        :param services: services to be added
+        :type services: iterable
+
+        """
+
+        if services:
+            self._add_svcs.update(services)
+
+    def add_ports(self, ports):
+        """
+        Ports that should be allowed through firewall.
+
+        :param ports: ports to be added
+        :type ports: iterable
+
+        """
+
+        if ports:
+            self._add_ports.update(ports)
+
+    def add_trusts(self, trusts):
+        """
+        trusts that should be allowed through firewall.
+
+        :param trusts: trusts to be added
+        :type trusts: iterable
+
+        """
+
+        if trusts:
+            self._add_trusts.update(trusts)
+
+    def remove_services(self, services):
+        """
+        New services that should not be allowed through firewall.
+
+        :param services: services to be removed
+        :type services: iterable
+
+        """
+
+        if services:
+            self._remove_svcs.update(services)
+
+    def firewall_enabled(self, fwenabled):
+        """Enable or disable firewall"""
+
+        if fwenabled is not None:
+            self._firewall_enabled = fwenabled
+
+    def __str__(self):
+        """Standard method useful for debugging and testing."""
+
+        ret = "firewall"
```

Comment:
True. We should make sure that this is done with the refactoring for the latest version of Anaconda. Right now, this is to maintain code consistency throughout.

Reply:
I don't know, the code that would use string components in a list would not be only faster, but also shorter - I took another look and you are forced to join iterables into strings, which would become unnecessary.
I don't see this as a blocker though - you are right that it can be refactored any time.